apis/addons/v1alpha1/addons_types.go

```diff
@@ -486,6 +491,21 @@ func (a *Addon) UpgradeCompleteForCurrentVersion() bool {
 		a.Status.UpgradePolicy.Value == AddonUpgradePolicyValueCompleted
 }
 
+// Gets the AddonInstallOLMCommon by Install Type
+func (a *Addon) GetAddonInstallOLMCommon() (AddonInstallOLMCommon, error) {
+	switch a.Spec.Install.Type {
+	case OLMAllNamespaces:
+		if a.Spec.Install.OLMAllNamespaces != nil {
+			return a.Spec.Install.OLMAllNamespaces.AddonInstallOLMCommon, nil
+		}
+	default:
```

Comment:
Just a small thing we can keep it as " case addonsv1alpha1.OLMOwnNamespace: "  . Even though I see only two cases .

Reply:
@venkateshsredhat Did you mean no default and just return an error if the install type is not supported?

Reply:
@pepedocs  I see currently we handle only two cases here https://github.com/openshift/addon-operator/blob/main/internal/controllers/addon/phase_ensure_subscription.go#L38 . So I would want this function to be similar to that .

Reply:
@venkateshsredhat gotcha! In that code you refer though, it didn't handle no matched case and looks like it assumed it will always match. I'm going to update my commit but I'll also add the error part if there is not case match. Please confirm.

Reply:
@pepedocs  yup , add this case and another default statement to handle error .

Reply:
@venkateshsredhat Thanks for confirming. I've removed the default case and used a specific case instead. A no-match will cause to return an error.